lib/metaclient/meta_client.go

```diff
@@ -3353,6 +3354,50 @@ func (c *Client) registerOffset(currentServer int, host string) (uint64, error)
 	return callback.Offset, nil
 }
 
+func (c *Client) ThermalShards(dbName string, start, end time.Duration) map[uint64]struct{} {
+	startTime := time.Now()
+	shards := make(map[uint64]struct{})
+	var lt, rt time.Time
+
+	if start != 0 {
+		lt = time.Now().Add(-start)
```

Comment:
Please check if it is necessary to convert time to UTC time due to startTime and endTime of shardgroup is UTC time.